src/checkers/inference/solver/frontend/TwoQualifiersLattice.java

```diff
@@ -0,0 +1,54 @@
+package checkers.inference.solver.frontend;
+
+import org.checkerframework.javacutil.AnnotationUtils;
+
+import java.util.Set;
+
+import javax.lang.model.element.AnnotationMirror;
+
+/**
+ * Special Lattice class for two qualifier type system.
+ * 
+ * @author jianchu
+ *
+ */
+public class TwoQualifiersLattice extends Lattice {
+
+    protected Set<AnnotationMirror> allTypes;
```

Comment:
In super class `Lattice` we have a `allTypes` in type of `Set<? extends AnnotationMirror>`. What is the reason of declaring `allTypes` with type `Set<AnnotationMirror>` here? It seems this one has the same meaning with the `allTypes` in the super class, except the type are different. So to me it is like a duplication.